bindings/rust/src/lib.rs

```diff
@@ -1064,4 +1065,154 @@ impl<'a, D> Unicorn<'a, D> {
+        if err == uc_error::OK {
+            Ok(())
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_get_arch(&self) -> Result<Arch, uc_error> {
+        let mut result: i32 = Default::default();
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_ARCH | UC_CTL_IO_READ, &mut result) };
+        if err == uc_error::OK {
+            Arch::try_from(result as usize)
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_get_timeout(&self) -> Result<u64, uc_error> {
+        let mut result: u64 = Default::default();
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_ARCH | UC_CTL_IO_READ, &mut result) };
+        if err == uc_error::OK {
+            Ok(result)
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_exits_enable(&self) -> Result<(), uc_error> {
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_USE_EXITS | UC_CTL_IO_WRITE, 1) };
+        if err == uc_error::OK {
+            Ok(())
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_exits_disable(&self) -> Result<(), uc_error> {
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_USE_EXITS | UC_CTL_IO_WRITE, 0) };
+        if err == uc_error::OK {
+            Ok(())
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_get_exits_count(&self) -> Result<usize, uc_error> {
+        let mut result: libc::size_t = Default::default();
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_EXITS_CNT | UC_CTL_IO_READ, &mut result) };
+        if err == uc_error::OK {
+            Ok(result)
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_get_exits(&self) -> Result<&[u64], uc_error> {
+        let exits_count: libc::size_t = self.ctl_get_exits_count()?;
+        let mut buffer = unsafe { libc::malloc(exits_count * 8) };
+        if buffer.is_null() {
+            return Err(uc_error::NOMEM);
+        }
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_EXITS | UC_CTL_IO_READ, &mut buffer, exits_count) };
+        if err == uc_error::OK {
+            let slice: &[u64] = unsafe { std::slice::from_raw_parts(std::mem::transmute::<*mut c_void, *const u64>(buffer), exits_count) };
+            let slice_owned = slice.clone();
+            unsafe { libc::free(buffer) };
+            Ok(slice_owned)
+        } else {
+            unsafe { libc::free(buffer) };
+            Err(err)
+        }
+    }
+
+    pub fn ctl_set_exits(&self, exits: Vec<*mut c_void>) -> Result<(), uc_error> {
+        let err = unsafe { ffi::uc_ctl(self.get_handle(), UC_CTL_UC_EXITS | UC_CTL_IO_WRITE, exits.as_ptr(), exits.len() as libc::size_t) };
+        if err == uc_error::OK {
+            Ok(())
+        } else {
+            Err(err)
+        }
+    }
+
+    pub fn ctl_get_cpu_model(&self) -> Result<i32, uc_error> {
```

Comment:
Would be nice to had some explicit type for the CPU. I know this is not easy because the values of different architectures overlap.